lmod_proxy/static/site/js/site.js

```diff
@@ -0,0 +1,129 @@
+$(document).ready(function(){
+
+  // Setup nice click handling for gradebook
+  $("#gradebook").click(function() {
+    if(!$(this).hasClass("selected")) {
+      $(this).select();
+      $(this).addClass("selected");
+    }
+  });
+  $("#gradebook").blur(function() {
+    if($(this).hasClass("selected")) {
+      $(this).removeClass("selected");
+    }
+  });
+
+
+  function post_call(data) {
+    $('#error').hide(100);
+    console.log(data)
+    $.post(location.pathname, data)
+      .done(function(data, status){
+        console.log(data);
+        console.log(status);
+        $('#result-msg').html(data['msg']);
+        var tbl=$("<table/>").attr("id","datatable");
+        var row = "<thead><tr>";
+        if(typeof data['data'][0] !== 'undefined') {
+          $.each(Object.keys(data['data'][0]), function(index, item) {
+            row += "<td>" + item + "</td>";
+          });
+          row += "</tr></thead>";
+          tbl.append(row);
+          tbl.append('<tbody');
+          var body = '';
+          $.each(data['data'], function(row_index, row_item) {
+            body += '<tr>';
+            $.each(row_item, function(item_index, item) {
+              body += '<td>' + item + '</td>';
+            });
+            body += '<tr>';
+          });
+          tbl.append(body);
```

Comment:
I definitely like it, and $.map and item in the callback makes more sense than $.each(item) as suggested above.  I didn't know about .map and that you were suggesting it. Excuse my ignorance